lib/jobs/dell-wsman-configure-bios.js

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017, DELL, Inc.
+
+'use strict';
+
+var di = require('di'),
+    urlParse = require('url-parse');
+
+module.exports = DellWsmanConfigureBiosFactory;
+di.annotate(DellWsmanConfigureBiosFactory, new di.Provide('Job.Dell.Wsman.ConfigureBios'));
+di.annotate(DellWsmanConfigureBiosFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Promise',
+    'Assert',
+    'Util',
+    'Services.Waterline',
+    'Services.Configuration',
+    '_',
+    'HttpTool',
+    'Errors',
+    'Services.Encryption',
+    'validator'
+));
+
+function DellWsmanConfigureBiosFactory(
+    BaseJob,
+    Logger,
+    Promise,
+    assert,
+    util,
+    waterline,
+    configuration,
+    _,
+    HttpTool,
+    errors,
+    encryption,
+    validator
+) {
+    var logger = Logger.initialize(DellWsmanConfigureBiosFactory);
+
+    function DellWsmanConfigureBiosJob(options, context, taskId) {
+        DellWsmanConfigureBiosJob.super_.call(this, logger, options, context, taskId);
+        assert.object(this.options);
+        this.nodeId = this.context.target;
+    }
+
+    util.inherits(DellWsmanConfigureBiosJob, BaseJob);
+
+    DellWsmanConfigureBiosJob.prototype._run = function () {
+        var self = this;
+        self.dell = configuration.get('dell');
+
+        if (!self.dell || !self.dell.services || !self.dell.services.configuration) {
```

Comment:
A trivial style suggestion, if you design a function to return promise, then it's better to wrap all error into Promise, so the caller can use promise only to catch all errors, rather than use both promise catch and `try/catch`